### ISSUE TITLE
No option to make boxplot box colour transparent

### ISSUE THREAD

**Issue by lydiayliu** [uclahs-cds/public-R-BoutrosLab-plotting-general#34]
As far as I can tell, there is no option to make the box colour in boxplot transparent (aka there is no `alpha` option availalbe, only `col`). The obvious downside to this is that if there is dots / stripplot added, the dots in the box are completely blocked by the colour of the box.

Example from API:
```
reformatted.data <- data.frame(
    x <- as.vector(t(microarray[1:10,1:58])),
    y <- as.factor(rep(rownames(microarray[1:10,1:58]),each = 58))
    );

create.boxplot(
    #
    formula = y ~ x,
    data = reformatted.data,
    main = 'Legend',
    xaxis.cex = 1,
    yaxis.cex = 1,
    xlab.cex = 1.5,
    ylab.cex = 1.5,
    ylab.label = 'Gene',
    xlimits = c(0,13),
    xat = seq(0,12,2),
    col = c('dodgerblue', 'pink'),
    add.stripplot = TRUE,
    description = 'Boxplot created by BoutrosLab.plotting.general'
    );
```

<img width="693" alt="image" src="https://user-images.githubusercontent.com/10492815/183541565-77df382e-391a-48c4-b341-54e39f029a62.png">

My solution to this has always been to create my own colours with alpha "built-in"

```
adjust_colour_alpha <- function(color, percent = 50, name = NULL) {

    ## Get RGB values for named color
    rgb.val <- col2rgb(color)

    ## Make new color using input color as base and alpha set by transparency
    t.col <- rgb(rgb.val[1], rgb.val[2], rgb.val[3],
                 max = 255,
                 alpha = percent * 255 / 100,
                 names = name)

    return(t.col)
}

reformatted.data <- data.frame(
    x <- as.vector(t(microarray[1:10,1:58])),
    y <- as.factor(rep(rownames(microarray[1:10,1:58]),each = 58))
    );

create.boxplot(
    #
    formula = y ~ x,
    data = reformatted.data,
    main = 'Legend',
    xaxis.cex = 1,
    yaxis.cex = 1,
    xlab.cex = 1.5,
    ylab.cex = 1.5,
    ylab.label = 'Gene',
    xlimits = c(0,13),
    xat = seq(0,12,2),
    col = sapply(c('dodgerblue', 'pink'), adjust_colour_alpha),
    add.stripplot = TRUE,
    description = 'Boxplot created by BoutrosLab.plotting.general'
    );
```

<img width="773" alt="image" src="https://user-images.githubusercontent.com/10492815/183541698-880409b7-d875-43cc-b2f9-b6ce5d0a2c19.png">

But this can be a function of BPG instead :)

**Comment by stefaneng** [comment]
@lydiayliu I fixed this in PR #41. In the future if you need to make any color transparent you can use [`grDevices::adjustcolor`](https://www.rdocumentation.org/packages/grDevices/versions/3.6.2/topics/adjustcolor) which is already imported by lattice so will always be available from BPG.

**Comment by lydiayliu** [comment]
Coooooool good to know! Thank you!
